UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Services/EssWebhookService.cs

```diff
@@ -55,37 +79,109 @@ public async Task DeleteSearchAndDownloadCacheData(EnterpriseEventCacheDataReque
-
-                    logger.LogInformation(EventIds.DeleteSearchDownloadCacheDataFromContainerStarted.ToEventId(), "Deletion started for Search and Download cache data from Blob Container for ProductName:{cellName} of BusinessUnit:{businessUnit} and BatchId:{cacheTableData.BatchId} and _X-Correlation-ID:{CorrelationId}", cellName, enterpriseEventCacheDataRequest.BusinessUnit, cacheTableData.BatchId, correlationId);
-                    await azureBlobStorageClient.DeleteCacheContainer(storageConnectionString, cacheTableData.BatchId);
-                    logger.LogInformation(EventIds.DeleteSearchDownloadCacheDataFromContainerCompleted.ToEventId(), "Deletion completed for Search and Download cache data from Blob Container for ProductName:{cellName} of BusinessUnit:{businessUnit} and BatchId:{cacheTableData.BatchId} and _X-Correlation-ID:{CorrelationId}", cellName, enterpriseEventCacheDataRequest.BusinessUnit, cacheTableData.BatchId, correlationId);
+                    BatchId = enterpriseEventCacheDataRequest.BatchId,
+                    PartitionKey = cellName,
+                    RowKey = $"{editionNumber}|{updateNumber}|{enterpriseEventCacheDataRequest.BusinessUnit.ToUpper()}",
+                    Response = JsonConvert.SerializeObject(enterpriseEventCacheDataRequest)
+                };
+
+                await DeleteSearchAndDownloadCacheData(fssSearchResponse, storageConnectionString, correlationId);
+                if (enterpriseEventCacheDataRequest.Files.Count > 0 && enterpriseEventCacheDataRequest.Files != null)
+                {
+                    await CacheSearchAndDownloadData(fssSearchResponse, correlationId);
                 }
                 else
                 {
-                    logger.LogInformation(EventIds.DeleteSearchDownloadCacheNoDataFoundEvent.ToEventId(), "No Matching Product found in Search and Download Cache table:{cacheConfiguration.Value.FssSearchCacheTableName} with ProductName:{cellName} and BusinessUnit:{businessUnit} and _X-Correlation-ID:{CorrelationId}", cacheConfiguration.Value.FssSearchCacheTableName, cellName, enterpriseEventCacheDataRequest.BusinessUnit, correlationId);
+                    logger.LogInformation(EventIds.CacheSearchAndDownloadInvalidData.ToEventId(), "Cache search and download files data missing in Request for ProductName:{cellName}, BusinessUnit:{businessUnit} and ProductCode:{productCode} and _X-Correlation-ID:{CorrelationId}", cellName, enterpriseEventCacheDataRequest.BusinessUnit, productCode, correlationId);
```

Comment:
Can we change the EventName please as over here the data is missing not invalid.

Reply:
resolved